lms/djangoapps/courseware/course_tools.py

```diff
@@ -0,0 +1,73 @@
+"""
+Platform plugins to support a verified upgrade tool.
+"""
+
+import datetime
+import pytz
+from django.utils.translation import ugettext as _
+
+from course_modes.models import CourseMode
+from openedx.features.course_experience.course_tools import CourseTool
+from student.models import CourseEnrollment
+from courseware.date_summary import verified_upgrade_deadline_link
+from request_cache import get_request
+
+
+class VerifiedUpgradeTool(CourseTool):
+    """
+    The verified upgrade tool.
+    """
+    @classmethod
+    def analytics_id(cls):
+        """
+        Returns an id to uniquely identify this tool in analytics events.
+        """
+        return 'edx.tool.verified_upgrade'
+
+    @classmethod
+    def is_enabled(cls, request, course_key):
+        """
+        Show this tool to all learners who are eligible to upgrade.
+        """
+        enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
+        if enrollment is None:
+            return False
+
+        if enrollment.dynamic_upgrade_deadline is None:
+            return False
+
+        if not enrollment.is_active:
+            return False
+
+        if enrollment.mode not in CourseMode.UPSELL_TO_VERIFIED_MODES:
+            return False
+
+        if enrollment.course_upgrade_deadline is None:
+            return False
+
+        if datetime.datetime.now(pytz.UTC) >= enrollment.course_upgrade_deadline:
+            return False
+
+        return True
+
+    @classmethod
+    def title(cls):
+        """
+        Returns the title of this tool.
+        """
+        return _('Upgrade to Verified')
```

Comment:
@shamck This will appear in the sidebar under "bookmarks", is this text good?

![screenshot from 2017-10-18 15-46-53](https://user-images.githubusercontent.com/627177/31739361-9abadd8c-b41b-11e7-971f-fbdf86b410fb.png)

Reply:
Text looks good.